source/src_lcao/LCAO_gen_fixedH.cpp

```diff
@@ -1135,6 +1143,214 @@ void LCAO_gen_fixedH::build_Nonlocal_beta_new() //update by liuyu 2021-04-07
     ModuleBase::TITLE("LCAO_gen_fixedH","build_Nonlocal_beta_new");
     ModuleBase::timer::tick ("LCAO_gen_fixedH","build_Nonlocal_beta_new");
 
+#ifdef __MKL
+    const int mkl_threads = mkl_get_max_threads();
+    mkl_set_num_threads(1);
```

Comment:
MKL would by default run single-threaded if called from inside an active parallel region in the user code. There's no need to do so.

Reply:
If you intend to do so, please follow up [example](https://www.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-c/top/support-functions/threading-control/mkl-set-num-threads-local.html#mkl-set-num-threads-local_EXAMPLE) provided by intel.

Reply:
For MKL behavior in OpenMP parallel region, please read [this post](https://www.intel.com/content/www/us/en/develop/documentation/onemkl-linux-developer-guide/top/managing-performance-and-memory/improving-performance-with-threading/using-additional-threading-control/mkl-dynamic.html). Generally, such consideration is not needed.